merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ThreadedTask.java

```diff
@@ -46,8 +49,10 @@ public TaskStatus step(final Scheduler scheduler) {
       if (response instanceof TaskResponse.Success successResponse) {
         final var status = successResponse.status;
 
-        if (status instanceof TaskStatus.Completed) {
+        if (status instanceof TaskStatus.Completed completed) {
           this.lifecycle = Lifecycle.Terminated;
+          // SAFETY: This TaskStatus is constructed by this class, using the same ReturnType.
+          this.returnValue = (ReturnType) completed.returnValue();
```

Comment:
It's not clear how this cast is legal. The type of `TaskStatus.Completed#returnValue` is `TaskReturnValue<T>` for some `T`. By chasing the `SAFETY` clause, `T` is `ReturnType`, but this cast would require that `TaskReturnValue<ReturnType> = ReturnType`, which is unlikely?

Reply:
Yikes, that doesn't seem right, looks like I overlooked ThreadedTask on one of my many rebases. I'm surprised it didn't blow up during testing 🤔 

I don't quite remember why it's being saved in a field - I think for the purposes of this PR (return values in sim results only), there shouldn't be a need to cache it in a field. Taking a look now.

Reply:
By adding a generic to TaskResponse, this cast is no longer necessary :+1: